### PR TITLE
Adding tests for valid/invalid patterns

### PR DESCRIPTION
**Pull request by pmpurifoy**

### CODE REVIEW

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 1]
```diff
@@ -1288,6 +1288,34 @@ bool VerifyRemoveAllAsync()
     return true;
 }
 
+
+bool VerifyForegroundRegistrationSuccess()
+{
+    winrt::AppNotificationManager::Default().Unregister();
```
Comment (sharath2727) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:1294]: do you need a test if you call unregisterAll and then register notificationinvoked?

**Review by sharath2727 — APPROVED** [review]

File: test/TestApps/ToastNotificationsTestApp/main.cpp (hunk 1 above)
Comment (pmpurifoy) [reply test/TestApps/ToastNotificationsTestApp/main.cpp:1294]: The heart of the test is to register foreground handlers before you register. Unregister/UnregisterAll are interchangable.

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 2]
```diff
@@ -1288,6 +1288,34 @@ bool VerifyRemoveAllAsync()
...
+        winrt::event_token token{ winrt::AppNotificationManager::Default().NotificationInvoked([](const auto&, winrt::AppNotificationActivatedEventArgs const& toastArgs) {}) };
+    }
+    catch (...)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool VerifyForegroundRegistrationFails()
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:1306]: Please either write a comment specifying why this is expected to fail or provide a more descriptive test name.

File: test/ToastNotificationTests/APITests.cpp [hunk 3]
```diff
@@ -578,5 +578,14 @@ namespace Test::ToastNotifications
         {
             RunTestUnpackaged(L"VerifyToastUpdateZeroSequenceFail_Unpackaged", testWaitTime());
         }
+
+        TEST_METHOD(VerifyForegroundRegistrationSuccess)
+        {
+            RunTest(L"VerifyForegroundRegistrationSuccess", testWaitTime());
```
Comment (danielayala94) [inline test/ToastNotificationTests/APITests.cpp:584]: What about the unpackaged scenario tests?

File: test/TestApps/ToastNotificationsTestApp/main.cpp (hunk 2 above)
Comment (robertzhou-wpn) [reply test/TestApps/ToastNotificationsTestApp/main.cpp:1306]: +1

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 4]
```diff
@@ -1288,6 +1288,34 @@ bool VerifyRemoveAllAsync()
     return true;
 }
 
+
+bool VerifyForegroundRegistrationSuccess()
```
Comment (robertzhou-wpn) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:1292]: nit: looks like this is to verify the foreground registration succeed after un-registration. Maybe we can make this name of this test more descriptive.

File: test/ToastNotificationTests/APITests.cpp (hunk 3 above)
Comment (pmpurifoy) [reply test/ToastNotificationTests/APITests.cpp:584]: Unpackaged/Packaged have the same behavior. But I can add an unpackaged test.

File: test/TestApps/ToastNotificationsTestApp/main.cpp [hunk 5]
```diff
@@ -1303,7 +1304,8 @@ bool VerifyForegroundRegistrationSuccess()
     return true;
 }
 
-bool VerifyForegroundRegistrationFails()
+// Registering a foregorund handler after Register() is an invalid pattern for developers. Test must fail.
```
Comment (danielayala94) [inline test/TestApps/ToastNotificationsTestApp/main.cpp:1307]: typo nit: foreground

**Review by danielayala94 — APPROVED** [review]

**Comment by danielayala94** [comment]
What is the status of this PR?

**Review by raghavthind2005 — APPROVED** [review]

**Pull request closed without merging**